provisioning/baremetal_pod.go

```diff
@@ -553,12 +631,19 @@ func newMetal3PodTemplateSpec(images *Images, config *metal3iov1alpha1.Provision
 		},
 	}
 
+	volumes := []corev1.Volume{}
+	volumes = append(volumes, metal3Volumes...)
+	// Include the volumes needed by createKubeProxyContainer,
+	// used to set up the proxy container for metric
+	// collection.
+	volumes = append(volumes, newRBACConfigVolumes()...)
```

Comment:
nit: What about putting all the volumes related enrichment in something like `newMetal3Volumes()`

Reply:
Done.